runtime/queries/zig/highlights.scm

```diff
@@ -0,0 +1,277 @@
+(
+  [
+    ; union(Tag){}
+    (ContainerDeclType (SuffixExpr (IDENTIFIER) @type))
+
+    ; enum{}
+    (ContainerDeclType "enum")
+  ]
+  (ContainerField (IDENTIFIER) @constant)?
+)
+
+; INFO: .IDENTIFIER is a field?
+(SuffixExpr 
+  "."
+  (IDENTIFIER) @property
+)
+
+; error.OutOfMemory;
+(SuffixExpr 
+  "error"
+  "."
+  (IDENTIFIER) @constant
+)
+
+(VarDecl
+  (IDENTIFIER) @type
+  [
+    ; const IDENTIFIER = struct/enum/union...
+    (SuffixExpr (ContainerDecl))
+
+    ; const A = u8;
+    (SuffixExpr (BuildinTypeExpr))
+  ]
+)
+
+; const fn_no_comma = fn (i32, i32) void;
+(VarDecl
+  (IDENTIFIER) @function
+  (SuffixExpr (FnProto))
+)
+
+; var x: IDENTIFIER
+type: (SuffixExpr (IDENTIFIER) @type)
+
+; IDENTIFIER{}
+constructor: (SuffixExpr (IDENTIFIER) @constructor)
+
+;{.IDENTIFIER = 1}
+(FieldInit (IDENTIFIER) @property)
+
+; var.field
+(SuffixOp (IDENTIFIER) @property)
+
+; var.func().func().field
+( 
+  (SuffixOp
+    (IDENTIFIER) @function
+  )
+  .
+  (FnCallArguments)
+)
+; func()
+( 
+  (
+    (IDENTIFIER) @function
+  )
+  .
+  (FnCallArguments)
+)
+
+; functionn decl
+(FnProto
+  (IDENTIFIER) @function
+  (SuffixExpr (IDENTIFIER) @type)?
+  ("!")? @function.macro
+)
+
+(ParamDecl 
+  (ParamType (SuffixExpr (IDENTIFIER) @parameter))
+)
+
+(ParamDecl 
+  (IDENTIFIER) @parameter
+  ":"
+  [
+    (ParamType (SuffixExpr (IDENTIFIER) @type))
+    (ParamType)
+  ]
+)
+
+(SwitchItem 
+  (SuffixExpr
+    "."
+    .
+    (IDENTIFIER) @constant
+  )
+)
+
+(BUILTINIDENTIFIER) @function.builtin
+
+((BUILTINIDENTIFIER) @include
+  (#any-of? @include "@import" "@cImport"))
+
+(INTEGER) @number
+
+(FLOAT) @number
+
+[
+  (STRINGLITERAL)
+  (STRINGLITERALSINGLE)
+] @string
+
+(CHAR_LITERAL) @character
+
+[
+  "allowzero"
+  "volatile"
+  "anytype"
+  "anyframe"
+  (BuildinTypeExpr)
+] @type.builtin
+
+(BreakLabel (IDENTIFIER) @label)
+(BlockLabel (IDENTIFIER) @label)
+
+[
+  "true"
+  "false"
+] @constant.builtin
+
+[
+  "undefined"
+  "unreachable"
+  "null"
+] @constant.builtin
+
+[
+  "else"
+  "if"
+  "switch"
+] @keyword.control
+
+[
+  "for"
+  "while"
+] @keyword.control
+
+[
+  "or"
+  "and"
+  "orelse"
+] @operator
+
+[
+  "struct"
+  "enum"
+  "union"
+  "error"
+  "packed"
+  "opaque"
+] @keyword
+
+[
+  "try"
+  "error"
+  "catch"
+] @function.macro
+
+; VarDecl @keyword.function
+[
+  "const"
+  "var"
+  "comptime"
+  "threadlocal"
+  "fn"
+] @keyword.directive
+
+[
+  "test"
+  "pub"
+  "usingnamespace"
+] @keyword
+
+[
+  "return"
+  "break"
+  "continue"
+] @keyword
+
+; Macro
+[
+  "defer"
+  "errdefer"
+  "async"
+  "nosuspend"
+  "await"
+  "suspend"
+  "resume"
+  "export"
+  "extern"
+] @function.macro
+
+; PrecProc
+[
+  "inline"
+  "noinline"
+  "asm"
+  "callconv"
+  "noalias"
+] @attribute
+
+[
+  "linksection"
+  "align" 
+] @function.builtin
+
+[
+  (CompareOp)
+  (BitwiseOp)
+  (BitShiftOp)
+  (AdditionOp)
+  (MultiplyOp)
+  (PrefixOp)
+  "*"
+  "**"
+  "->"
+  "=>"
+  ".?"
+  ".*"
+  "="
+] @operator
```

Comment:
These should also be grouped up with the other `@operator` blocks into one `@operator` block.